### PR TITLE
feat(api): execute ascode action

### PR DESCRIPTION
**Pull request by sguiheux**
1. Description
1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#18709.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔ 
* Package
  * Docker ✘

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#18713.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✘

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#18714.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔ 
* Package
  * Docker ✘

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#18717.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✘

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#18718.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔ 
* Package
  * Docker ✘

File: engine/api/v2_repository_analyze.go [hunk 1]
```diff
@@ -425,6 +425,9 @@ func (api *API) analyzeRepository(ctx context.Context, projectRepoID string, ana
 	analysis.Data.Error = strings.Join(skippedFiles, "\n")
 	if len(skippedFiles) == len(analysis.Data.Entities) {
 		analysis.Status = sdk.RepositoryAnalysisStatusSkipped
+		if len(analysis.Data.Entities) == 0 {
+			analysis.Data.Error = "no files found"
```
Comment (yesnault) [inline engine/api/v2_repository_analyze.go:429]:
```suggestion
			analysis.Data.Error = "no file found"
```

File: sdk/exportentities/step.go [hunk 2]
```diff
@@ -363,6 +371,7 @@ type Step struct {
 	Checkout         *StepCheckout         `json:"checkout,omitempty" yaml:"checkout,omitempty" jsonschema:"oneof_required=actionCheckout" jsonschema_description:"Checkout repository for an application.\nhttps://ovh.github.io/cds/docs/actions/builtin-checkoutapplication"`
 	InstallKey       *StepInstallKey       `json:"installKey,omitempty" yaml:"installKey,omitempty" jsonschema:"oneof_required=actionInstallKey" jsonschema_description:"Install a key (GPG, SSH) in your current workspace.\nhttps://ovh.github.io/cds/docs/actions/builtin-installkey"`
 	Deploy           *StepDeploy           `json:"deploy,omitempty" yaml:"deploy,omitempty" jsonschema:"oneof_required=actionDeploy" jsonschema_description:"Deploy an application.\nhttps://ovh.github.io/cds/docs/actions/builtin-deployapplication"`
+	AsCodeAction     *StepAscodeAction     `json:"AsCodeAction,omitempty" yaml:"AsCodeAction,omitempty" jsonschema:"oneof_required=AsCodeAction" jsonschema_description:"ascode action"`
```
Comment (yesnault) [inline sdk/exportentities/step.go:374]:
```suggestion
	AsCodeAction     *StepAscodeAction     `json:"asCodeAction,omitempty" yaml:"asCodeAction,omitempty" jsonschema:"oneof_required=asCodeAction" jsonschema_description:"ascode action"`
```

File: sdk/workflowv3/step.go [hunk 3]
```diff
@@ -25,6 +25,7 @@ type Step struct {
 	InstallKey                *exportentities.StepInstallKey       `json:"installKey,omitempty" yaml:"installKey,omitempty"`
 	Deploy                    *exportentities.StepDeploy           `json:"deploy,omitempty" yaml:"deploy,omitempty"`
 	Promote                   *exportentities.StepPromote          `json:"promote,omitempty" yaml:"promote,omitempty"`
+	AsCodeAction              *exportentities.StepAscodeAction     `json:"AsCodeAction,omitempty" yaml:"AsCodeAction,omitempty"`
```
Comment (yesnault) [inline sdk/workflowv3/step.go:28]:
```suggestion
	AsCodeAction              *exportentities.StepAscodeAction     `json:"asCodeAction,omitempty" yaml:"asCodeAction,omitempty"`
```

**Review by yesnault — CHANGES_REQUESTED** [review]

File: tests/fixtures/ITCLIPRJVCS/pipeline.pip.yml [hunk 4]
```diff
@@ -1,9 +1,13 @@
...
+  - AsCode
 jobs:
-- job: test
+- job: AsCodeJob
+  stage: AsCode
   steps:
-  - script:
-    - env
+  - name: ITCLIPRJVCS/my_vcs_server/gituser/myrepo/test-action@master
+    AsCodeAction:
```
Comment (yesnault) [inline tests/fixtures/ITCLIPRJVCS/pipeline.pip.yml:10]:
```suggestion
    asCodeAction:
```

**Review by yesnault — CHANGES_REQUESTED** [review]

**Review by fsamin — APPROVED** [review]

**Pull request merged**